dashboard/config/libraries/NativeSpriteLab.interpreted.js

```diff
@@ -167,36 +170,106 @@ function setupSim(
+        newList.push(list[removeCounter]);
+      }
+    }
+    return newList;
+  }
+  function listIncludes(list, item) {
+    for (var includesCounter = 0; includesCounter < list.length; includesCounter++) {
+      if (list[includesCounter] == item) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  // Initialize sprites
   var counter_i = 0;
-  for (counter_i = 0; counter_i < s3number; counter_i++) {
-    makeNewSpriteAnon(s3costume, randomLocation());
+  for (counter_i = 0; counter_i < s1number; counter_i++) {
+    makeNewSpriteAnon(s1costume, randomLocation());
   }
-  setProp({costume: s3costume}, 'scale', 50);
+  addBehaviorSimple({costume: s1costume}, new Behavior(movementBehavior(s1speed)));
 
   for (counter_i = 0; counter_i < s2number; counter_i++) {
     makeNewSpriteAnon(s2costume, randomLocation());
   }
   addBehaviorSimple({costume: s2costume}, new Behavior(movementBehavior(s2speed)));
 
-  for (counter_i = 0; counter_i < s1number; counter_i++) {
-    makeNewSpriteAnon(s1costume, randomLocation());
+  for (counter_i = 0; counter_i < s3number; counter_i++) {
+    makeNewSpriteAnon(s3costume, randomLocation());
   }
-  addBehaviorSimple({costume: s1costume}, new Behavior(movementBehavior(s1speed)));
+  setProp({costume: s3costume}, 'scale', 50);
+
+
+  // We want to be able to randomize which sprite gets the point in the case of a tie. So the approach here is
+  // to use checkTouching() to detect all the collisions, but delay until the next frame to actually give a point
+  // and remove the s3 sprite. I'm using World.s1ToCollect and World.s2ToCollect to keep track of which sprite ids
+  // will be removed the following frame. Then if there's a tie (ie. s2 wants to remove a sprite id that's already
+  // in World.s1ToCollect or vice versa), use randomNumber(0,1) to give the sprite a 50-50 chance of "stealing" the
+  // s3 sprite.
+  // Then to actually remove the s3 sprites and keep track of the score, I'm using a behavior. I have the behavior
+  // attached to sprite id 0, but it doesn't really matter which sprite (as long as it's not an s3 sprite that
+  // might get deleted). A behavior is just the simplest way to add a snippet of code that will get executed each frame.
+  // The behavior basically just tallies the score and deletes the sprites from the previous frame, then checks
+  // whether the simulation should end. The one-frame delay isn't really noticeable since frames are so fast.
 
   checkTouching('while', {costume: s1costume}, {costume: s3costume}, function(extraArgs) {
-    destroy({id: extraArgs.target});
-    World.sprite1score++;
-    printText(s1costume + ' has collected ' + World.sprite1score);
-    checkSimulationEnd();
+    if (listIncludes(World.s1ToCollect, extraArgs.target)) {
+      // another s1 is also touching. Either way, s1 gets the point, but only count once, so don't add it again.
+    } else if (listIncludes(World.s2ToCollect, extraArgs.target)) {
```

Comment:
If there are two s1 touching s3 and one s2 touching s3, does this algorithm treat s1a, s1b, and s2a equally (each has 1/3 chance, so there is a 2/3 chance of an s1 winning) or do we just decide between s1 and s2 (each has 1/2 chance, so equal chance of either winning)? Maybe I'm not fully understanding, but I think it's the latter option - is that correct?

Reply:
Oh great catch. As it was, each s2 would have a 50% chance of stealing, resulting in a 75% chance of s2 getting the point (in the case where two s2 and one s1 are touching the same s3).

I reworked the alg so that in a tie, there's an equal chance of each individual sprite getting the s3 - so in the case where two s2 and one s1 are tied, you should expect a 2/3 chance of the s2 getting the point. Please take a look and let me know what you think!